### PR TITLE
Add counters to varnish input plugin

### PR DESCRIPTION
**Pull request by jamesmaidment**
Adds counters to varnish by parsing the JSON output of `varnishstat` for Gauge and Counter metric types.

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/varnish/varnish.go [hunk 1]
```diff
@@ -120,52 +120,72 @@ func (s *Varnish) Gather(acc telegraf.Accumulator) error {
...
-	for scanner.Scan() {
-		cols := strings.Fields(scanner.Text())
-		if len(cols) < 2 {
-			continue
+	sectionMap, err := parseVarnishJSON(out.Bytes(), s.filter)
+	if err != nil {
+		return fmt.Errorf("error parsing JSON output: %s", err)
+	}
+
+	for metricType, section := range sectionMap {
```
Comment (glinton) [inline plugins/inputs/varnish/varnish.go]: Seems like passing the accumulator to `parseVarnishJSON` would save a loop here

**Comment by danielnelson** [comment]
Need to do #3670 first to prevent this from being a breaking change.

**Comment by srebhan** [comment]
@jamesmaidment are you still interested in getting this merged?

**Comment by srebhan** [comment]
@jamesmaidment any chance you work on this?

**Comment by srebhan** [comment]
Is anyone willing to pick up this PR?

**Comment by sjwang90** [comment]
Closing this PR due to inactivity. If someone is interested on taking this work back up, please feel free to re-open the PR and take over the code.

**Pull request closed without merging**